stac_fastapi/types/stac_fastapi/types/stac.py

```diff
@@ -188,3 +153,78 @@ class PatchMoveCopy(StacBaseModel):
 
 
 PatchOperation = Union[PatchAddReplaceTest, PatchMoveCopy, PatchRemove]
+
+
+class BasePartial(StacBaseModel):
+    """Base Partial Class."""
+
+    @staticmethod
+    def merge_to_operations(data: Dict) -> List[PatchOperation]:
+        """Convert merge operation to list of RF6902 operations.
+
+        Args:
+            data: dictionary to convert.
+
+        Returns:
+            List: list of RF6902 operations.
+        """
+        operations = []
+
+        for key, value in data.copy().items():
+            if value is None:
+                operations.append(PatchRemove(op="remove", path=key))
+
+            elif isinstance(value, dict):
+                nested_operations = BasePartial.merge_to_operations(value)
+
+                for nested_operation in nested_operations:
+                    nested_operation.path = f"{key}/{nested_operation.path}"
+                    operations.append(nested_operation)
+
+            else:
+                operations.append(PatchAddReplaceTest(op="add", path=key, value=value))
+
+        return operations
+
+    def operations(self) -> List[PatchOperation]:
+        """Equivalent RF6902 operations to merge of Partial.
+
+        Returns:
+            List[PatchOperation]: Equivalent list of RF6902 operations
+        """
+        return self.merge_to_operations(self.model_dump())
+
+
+class PartialCollection(BasePartial):
+    """Partial STAC Collection."""
+
+    type: Optional[str]
```

Comment:
not sure why/how the test passes but for Optional we need to set `= None` with pydantic 2.0